docker/docker.go

```diff
@@ -0,0 +1,163 @@
+package docker
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"math/rand"
+	"net"
+
+	//"os"
+	"strconv"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/client"
+	"github.com/docker/go-connections/nat"
+
+	"github.com/dachad/tcpgoon/debugging"
+)
+
+type IPandPort struct {
+	IP   string
+	Port int
+}
+
+func lookForAvailableLocalPort() (availablePort int){
+    minPort := 1025
+	maxPort := 65535
+
+	for {
+		availablePort = rand.Intn(maxPort - minPort) + minPort
+		ln, err := net.Listen("tcp", ":" + strconv.Itoa(availablePort))
+		ln.Close()
+		if err == nil {
+			break
+		}
+	}
+	return availablePort
+}
+
+func DownloadAndRun(image string, port int) (targetinfo IPandPort, containerID string) {
```

Comment:
i guess this requires refactor / better encapsulation... long and with too much statements inside if/elses. I would also not panic in this function, but return an error and make the caller to manage the errors accordingly

Reply:
sure, do not take this PR as ready to review. The point here is, having a first working version, understand if this docker mode should be part of the regular command or have a custom one.